var/spack/repos/spack_repo/builtin/packages/py_zarr/package.py

```diff
@@ -49,8 +50,10 @@ class PyZarr(PythonPackage):
         depends_on("python@3.11:", type=("build", "run"))
         depends_on("py-hatchling", type="build")
         depends_on("py-hatch-vcs", type="build")
-        depends_on("py-packaging@:22.0", type=("build", "run"))
-        depends_on("py-numpy@0.14:", type=("build", "run"))
+        depends_on("py-packaging@22:", type=("build", "run"))
+        # numpy@2: in 2.18.0
+        # https://github.com/zarr-developers/zarr-python/issues/1818
+        depends_on("py-numpy@1.25:", type=("build", "run"))
```

Comment:
When is which `py-numpy` version required?
```suggestion
        depends_on("py-numpy@1.25:", type=("build", "run"), when="XYZ")
```

Reply:
it's within the `with when` block. 3.x line is entirely `py-numpy@1.25:`